### DISCUSSION TITLE
How to rollback from the rescue shell?

### DISCUSSION THREAD

**Discussion by MikiVanousek** [nix-community/nix-on-droid#266] [Q&A]
I wrote a configuration that causes an error when the shell starts. How can I roll back, if I only have access to the rescue shell? The `nix-on-droid` command is not available.

**Comment by t184256** [comment]
Hard to answer without knowing what exactly "causes an error when the shell starts" means and how hard is it broken.

Activation scripts for the past generations live in profile directories inside `/data/data/com.termux.nix/files/usr/nix/var/nix/profiles`. Executing them should happen from within the nix-on-droid environment though, entered using `/data/data/com.termux.nix/files/usr/bin/login`, which might or might not work. So the best suggestion I have without knowing the details would be something like `/data/data/com.termux.nix/files/usr/bin/login /data/data/com.termux.nix/files/usr/nix/var/nix/profiles/nix-on-droid-1-link/activate`.

**Comment by MikiVanousek** [comment]
This has helped me, but I still think it would be nice if there was a way to perform `nix-on-droid rollback` from the rescue shell.

**Reply by 573** [reply]
`/data/data/com.termux.nix/files/usr/bin/login /data/data/com.termux.nix/files/usr/nix/var/nix/profiles/nix-on-droid-1-link/activate` should do basically that assumed the last functional profile generation was used (i. e. replace `nix-on-droid-1-link`).

Also how to run `nix-on-droid` anyway when the assumption is a broken shell already ?

**Comment by t184256** [comment]
On August 15, 2023 2:38:44 PM GMT+02:00, Daniel Kahlenberg ***@***.***> wrote:
>`/data/data/com.termux.nix/files/usr/bin/login /data/data/com.termux.nix/files/usr/nix/var/nix/profiles/nix-on-droid-1-link/activate` should do basically that assumed the last functional profile generation was used (i. e. replace `nix-on-droid-1-link`).
>
>Also how to run `nix-on-droid` anyway when the assumption is a broken shell already ?

`/data/data/com.termux.nix/files/usr/bin/login nix-on-droid`, foregoing the interactive shell
